src/index.ts

```diff
@@ -86,23 +86,19 @@ export type Style = {
 const styles = new WeakMap<Element, Style>()
 
 export class MarkdownButtonElement extends HTMLElement {
-  constructor() {
-    super()
-    this.addEventListener('keydown', keydown(this.click))
-    this.addEventListener('click', this.click)
-  }
-
   connectedCallback() {
     if (!this.hasAttribute('role')) {
       this.setAttribute('role', 'button')
     }
+    this.addEventListener('keydown', keydown(this.click.bind(this)))
+    this.addEventListener('click', this.click.bind(this))
   }
 
   get styles() {
     return styles.get(this)
   }
 
-  click = () => {
+  click() {
```

Comment:
I believe if you're moving the setup call to `connectedCallback` you technically are free to leave this as `click = () => {}`.

I do generally agree though that I prefer the `.bind(this)` approach. (And frankly would prefer it even in our internal components.) For the purpose here it's much easier to expect the extending class to just write `click()`.

Reply:
Yup! This pr basically fixes the problem in two ways. I do think I just generally want to avoid functions as properties because of this somewhat unintuitive inheritance behavior.